g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductComponentAction.java

```diff
@@ -180,5 +180,84 @@ public APIResponseDTO checkTranslationResult(String productName, String componen
 						en);
 		}
 		return m;
-	}	
+	}
+
+	/**
+	 * Get the the matched version from the version list by comparing to the input version
+	 *
+	 * @param productName
+	 * @param version
+	 * @return a matched version, if there's no matched version then return input version
+	 */
+	private String getMatchedVersion(final String productName, final String version) throws L3APIException{
+		Map<String, String[]> productsAndVersions = productService.getProductsAndVersions();
+		String matchedVersion = "";
+		if(productsAndVersions != null && !productsAndVersions.isEmpty()) {
+			String[] versionList = productsAndVersions.get(productName);
+			if(versionList != null && versionList.length > 0) {
+				for(String s : versionList) {
+					String f = filterVersion(s, version);
+					if(!compare(f, version) && compare(s, matchedVersion)) {
+                        matchedVersion = s;
+					}
+				}
+			}
+		}
+		if(!StringUtils.isEmpty(matchedVersion)) {
+			return matchedVersion;
+		} else {
+			return version;
+		}
+	}
+
+	/**
+	 * Compare source version and target version
+	 *
+	 * @param source
+	 * @param target
+	 * @return true means source > target
+	 */
+	private boolean compare(final String source, final String target) {
```

Comment:
suggest to return 
-1 -> less than
0 -> equal
1 -》 greatter than

Reply:
agreed, this should be more clear.

Reply:
agreed, returning int should be more clear.